repo.go

```diff
@@ -519,36 +520,49 @@ func (r *Repo) setTopLevelMeta(roleFilename string, meta interface{}) error {
 	return r.local.SetMeta(roleFilename, b)
 }
 
-func (r *Repo) Sign(roleFilename string) error {
-	role := strings.TrimSuffix(roleFilename, ".json")
+// Use the keys associated with role to sign the payload in signed.
+func (r *Repo) SignPayload(role string, signed *data.Signed) (int, error) {
 	if !roles.IsTopLevelRole(role) {
-		return ErrInvalidRole{role, "only signing top-level metadata supported"}
+		return -1, ErrInvalidRole{role, "only signing top-level metadata supported"}
```

Comment:
Maybe use something better than -1?

Reply:
Happy to change this, but my understanding is that in Go it's [idiomatic to totally ignore the success-case return value](https://stackoverflow.com/questions/46726832/how-to-return-int-or-nil-in-golang) if an error is passed. In fact, I see [Go by Example](https://gobyexample.com/errors) does exactly this, with a `-1` return value.

What do you propose I do instead?

I've improved the documentation for this function to be clearer about what it returns in these cases.

Reply:
> and I wonder if having that number of keys returned is necessary? 

I think it might be useful for programmatic callers. I might want to double check that I've signed with the same number of keys that I expected.

> As far as I see we only use this value for logging a message, so we should be safe to return only the err value and log numKeys again whenever it is necessary from within the SignPayload() function. What are your thoughts on that?

I'd strongly prefer to avoid printing messages that we expect users to see within the library calls. Debug messages etc. are reasonable, but the output of the command should come from the command itself. Otherwise, applications might see our output in the middle of their operation.

So my vote is to leave it as-is. I would also be willing remove the number of keys from the output and the return value if you feel very strongly.

Reply:
> I think it might be useful for programmatic callers. I might want to double check that I've signed with the same number of keys that I expected.

I'm okay with keeping it then if there's such a use case 👍 

> I'd strongly prefer to avoid printing messages that we expect users to see within the library calls. Debug messages etc. are reasonable, but the output of the command should come from the command itself. Otherwise, applications might see our output in the middle of their operation.
> 
Agree 👍 Makes sense.